MSAL/src/native_auth/controllers/sign_in/MSALNativeAuthSignInController.swift

```diff
@@ -140,6 +141,12 @@ final class MSALNativeAuthSignInController: MSALNativeAuthTokenController, MSALN
                         self?.stopTelemetryEvent(telemetryInfo.event, context: context, delegateDispatcherResult: result)
                     }))
                 },
+                onAwaitingMFA: { _ in
```

Comment:
1. We don't need [weak self] in this case (and TBH I don't think we can use it). We use weak self in the telemetry callback from the delegate dispatcher.
2. Yes, this code is also used in the signIn after SSPR flow

Reply:
For 2, what I mean is that we are returning a `SignInAfterSignUpError` in case the user comes from the SignInAfter SSPR flow (instead of `SignInAfterResetPasswordError`).

Reply:
This is expected. The `SignInAfterSignUpError` is translated to `SignInAfterResetPasswordError` in the SSPR state class. But, I noticed that we use the wrong the telemetry id for the SSPR flow, so I'm going to fix it in this PR